src/binfhe/include/lwe-pke.h

```diff
@@ -71,6 +89,28 @@ class LWEEncryptionScheme {
     LWECiphertext Encrypt(const std::shared_ptr<LWECryptoParams> params, ConstLWEPrivateKey sk, const LWEPlaintext& m,
                           const LWEPlaintextModulus& p = 4, const NativeInteger& mod = 0) const;
 
+    /**
+   * Encrypts a bit using a public key (asymmetric key encryption)
+   *
+   * @param params a shared pointer to LWE scheme parameters
+   * @param pk - the secret key
+   * @param &m - the plaintext
+   * @param &p - the plaintext space
+   * @return a shared pointer to the ciphertext
+   */
+    LWECiphertext EncryptN(const std::shared_ptr<LWECryptoParams> params, ConstLWEPublicKey pk, const LWEPlaintext& m,
```

Comment:
passing m by const reference may seems as a good decision, but it is needless. LWEPlaintext and LWEPlaintextModulus are integers and should be passed by value. in this case you don't need to make them const and passing by value may be even faster.